### PR TITLE
Add check for TimeoutError in test/net/connect

### PR DESCRIPTION
**Pull request by AntonioLangiu**
When running this test on a machine with icmp packet blocked
the tests fails, because it receive a TimeoutError instead of a NetworkError

The problem is explained in #448

### CODE REVIEW

**Comment by AntonioLangiu** [comment]
Now the OSX build success!

File: test/net/connect.cpp [hunk 1]
```diff
@@ -110,14 +110,20 @@ TEST_CASE("connect_base works with ipv4") {
     });
 }
 
+int checkError (Error err) {
```
Comment (bassosimone) [inline test/net/connect.cpp]: Could you make this function static and name it using snake case?

File: test/net/connect.cpp [hunk 2]
```diff
@@ -110,14 +110,20 @@ TEST_CASE("connect_base works with ipv4") {
...
+    if (err == NetworkError() || err == TimeoutError())
+        return 1;
+    return 0;
```
Comment (bassosimone) [inline test/net/connect.cpp]: Could you more compactly write this as `return err == NetworkError() or err == TimeoutError();`?

File: test/net/connect.cpp [hunk 3]
```diff
@@ -110,14 +110,20 @@ TEST_CASE("connect_base works with ipv4") {
...
+
 TEST_CASE("connect_base works with ipv4 and closed port") {
     if (CheckConnectivity::is_down()) {
         return;
     }
     loop_with_initial_event([]() {
         connect_base("130.192.16.172", 81,
                 [](Error err, bufferevent *bev) {
-                    REQUIRE(err == NetworkError());
+                    REQUIRE(checkError (err));
```
Comment (bassosimone) [inline test/net/connect.cpp]: (There should not be space between a function name and the open brace)

**Comment by bassosimone** [comment]
Thank you! Will be happy to merge once the requested style fixes are applied!

**Comment by bassosimone** [comment]
Thanks!

**Comment by bassosimone** [comment]
Closes #448

**Pull request merged**